[BUG] http-password-checker rust CI is failing

### Affected project(s)

- [ ] documentation
- [x] examples
- [ ] wasmCloud host
- [ ] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Describe the bug

http-password-checker:rust CI is failing on main.

Seen in https://github.com/wasmCloud/wasmCloud/actions/runs/18019490008/job/51273737560?pr=4797

```
error[E0433]: failed to resolve: could not find `ContentVisitor` in `de`
  --> src/api.rs:14:57
   |
14 | #[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
   |                                                         ^^^^^^^^^^^^^^^^^^ could not find `ContentVisitor` in `de`
   |
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/serde-1.0.226/src/private/de.rs:14:75
   |
12 | #[cfg(any(feature = "std", feature = "alloc"))]
   |          ------------------------------------ the item is gated here
13 | pub use self::content::{
14 |     content_as_str, Content, ContentDeserializer, ContentRefDeserializer, ContentVisitor,
   |                                                                           ^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `ContentRefDeserializer` in `de`
  --> src/api.rs:14:57
   |
14 | #[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
   |                                                         ^^^^^^^^^^^^^^^^^^ could not find `ContentRefDeserializer` in `de`
   |
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/serde-1.0.226/src/private/de.rs:14:51
   |
12 | #[cfg(any(feature = "std", feature = "alloc"))]
   |          ------------------------------------ the item is gated here
13 | pub use self::content::{
14 |     content_as_str, Content, ContentDeserializer, ContentRefDeserializer, ContentVisitor,
   |                                                   ^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `TaggedContentVisitor` in `de`
  --> src/api.rs:53:57
   |
53 | #[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
   |                                                         ^^^^^^^^^^^^^^^^^^ could not find `TaggedContentVisitor` in `de`
   |
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/serde-1.0.226/src/private/de.rs:16:79
   |
12 | #[cfg(any(feature = "std", feature = "alloc"))]
   |          ------------------------------------ the item is gated here
...
16 |     TagContentOtherFieldVisitor, TagOrContentField, TagOrContentFieldVisitor, TaggedContentVisitor,
   |                                                                               ^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `ContentDeserializer` in `de`
  --> src/api.rs:53:57
   |
53 | #[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
   |                                                         ^^^^^^^^^^^^^^^^^^ could not find `ContentDeserializer` in `de`
   |
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/serde-1.0.226/src/private/de.rs:14:30
   |
12 | #[cfg(any(feature = "std", feature = "alloc"))]
   |          ------------------------------------ the item is gated here
13 | pub use self::content::{
14 |     content_as_str, Content, ContentDeserializer, ContentRefDeserializer, ContentVisitor,
   |                              ^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0433`.
error: could not compile `http-password-checker` (lib) due to 4 previous errors

Compiling component failed: exit status: 101
```

### Steps to reproduce

https://github.com/wasmCloud/wasmCloud/actions/runs/18019490008/job/51273737560?pr=4797

### Expected behavior

CI passes

### Environment

Failed on stable-x86_64-unknown-linux-gnu

### Screenshots / Logs / Additional context

_No response_

--- a/examples/rust/components/blobby/wit/deps/wasi-io-0.2.1/package.wit
+++ b/examples/rust/components/blobby/wit/deps/wasi-io-0.2.1/package.wit
@@ -1,5 +1,11 @@
 package wasi:io@0.2.1;
 
+interface error {
+  resource error {
+    to-debug-string: func() -> string;
+  }
+}
+
 interface poll {
   resource pollable {
     ready: func() -> bool;
@@ -7,12 +13,6 @@ interface poll {
   }
 
   poll: func(in: list<borrow<pollable>>) -> list<u32>;
-}
-
-interface error {
-  resource error {
-    to-debug-string: func() -> string;
-  }
 }
 
 interface streams {

--- a/examples/rust/components/http-password-checker/Cargo.toml
+++ b/examples/rust/components/http-password-checker/Cargo.toml
@@ -12,7 +12,7 @@ crate-type = ["cdylib"]
 anyhow = "1.0.92"
 # NOTE: While we'd *like* to use zxcvbn here, it is built strictly for browser-side WASM
 passwords = { version = "3.1.16", default-features = false }
-serde = { version = "1.0.214", default-features = false, features = ["derive"] }
+serde = { version = "1.0.214", default-features = false, features = ["derive", "alloc"] }
 serde_json = { version = "1.0.132", default-features = false, features = [ "std" ] }
 wasmcloud-component = "0.2.0"
 wit-bindgen = { version = "0.34" }

